### PR TITLE
ci: update github ci action

### PR DESCRIPTION
**Pull request by techytushar**
Signed-off-by: Tushar Mittal <chiragmittal.mittal@gmail.com>

Fixes: #63

### CODE REVIEW

**Comment by joewesch** [comment]
Hey @techytushar, thanks for the submission! Would you mind adding Python version 3.10 and Nautobot version 1.4 to the matrix as well?

**Comment by techytushar** [comment]
@joewesch Done ✅

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,13 +10,8 @@ jobs:
...
       matrix:
-        include:
-          - python-version: "3.7"
-            nautobot-version: "1.2"
-          - python-version: "3.8"
-            nautobot-version: "1.3"
-          - python-version: "3.9"
-            nautobot-version: "1.3"
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        nautobot-version: ["1.2", "1.3", "1.4"]
```
Comment (joewesch) [inline .github/workflows/ci.yml:14]: Looks like we need to exclude a combination that doesn't exist.

```suggestion
        nautobot-version: ["1.2", "1.3", "1.4"]
        exclude:
          - python-version: 1.2
            nautobot-version: 3.10
```

**Comment by jvanderaa** [comment]
@joewesch any thoughts on the failure within the tests? Do we just need to re-run it?

**Comment by joewesch** [comment]
The `nautobot-dev:1.2-py3.10` container tag does not exist, so we need to exclude it from the matrix. I added a suggestion that would do that.

**Comment by joewesch** [comment]
Now it's getting hit with the same bug as #79. @techytushar, sorry but can you remove Nautobot 1.4 from the tests for now. We will need to override the nautobot_config.py file if we want that version to run.

**Review by joewesch — APPROVED** [review]

**Pull request merged**